src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1104,6 +1103,155 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
+        flowHelperV2.addFlow(flow)
+
+        when: "Make the current path less preferable than alternatives"
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+        def alternativePaths = switchPair.paths.findAll { it != currentPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+
+        and: "Update the flow: port number and vlanId on the src/dst endpoints"
+        def updatedFlow = flow.jacksonCopy().tap {
+            it.source.portNumber = 10 // traffgen port
+            it.source.vlanId = flow.destination.vlanId + 1
+            it.destination.portNumber = 10 // traffgen port
+            it.destination.vlanId = flow.source.vlanId + 1
+        }
+        flowHelperV2.updateFlow(flow.flowId, updatedFlow)
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == updatedFlow.source.portNumber
+            it.source.vlanId == updatedFlow.source.vlanId
+        }
+
+        and: "Flow is not rerouted"
+        Wrappers.timedLoop(rerouteDelay + WAIT_OFFSET / 2) {
+            pathHelper.convert(northbound.getFlowPath(flow.flowId)) == currentPath
+        }
+
+        and: "System allows traffic on the flow"
+        def traffExam = traffExamProvider.get()
+        def examFlow = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(
+                flowHelperV2.toV1(updatedFlow), 100, 5
+        )
+        withPool {
+            [examFlow.forward, examFlow.reverse].eachParallel { direction ->
+                def resources = traffExam.startExam(direction)
+                direction.setResources(resources)
+                assert traffExam.waitExam(direction).hasTraffic()
+            }
+        }
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        and: "All involved switches pass switch validation"
+        withPool {
+            currentPath*.switchId.eachParallel { SwitchId swId ->
+                with(northbound.validateSwitch(swId)) { validation ->
+                    validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                    validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                }
+            }
+        }
+        def involvedSwitchesPassSwValidation = true
+
+        cleanup: "Revert system to original state"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        northbound.deleteLinkProps(northbound.getAllLinkProps())
+        !involvedSwitchesPassSwValidation && currentPath*.switchId.each { SwitchId swId ->
+            northbound.synchronizeSwitch(swId, true)
+        }
+        database.resetCosts()
+    }
+
+    @Tidy
+    def "System doesn't reroute flow to more preferable path during rollback after unsuccessfull update portNumber action"() {
```

Comment:
this scenario as-is seems to be useless, isl conflict validation is made prior the update happen, If you want to verify rollback you need to knock out src switch during the update for example, or make fl not receiving install commands

Reply:
deleted